### PR TITLE
remove backbone from auth

### PR DESCRIPTION
**Pull request by timmak**
In process of removing jQuery this also will require remove backbone rather make a few small PRs then on big one.

### CODE REVIEW

**Comment by garrensmith** [comment]
@timmak this looks great. I'll try and give a proper review later today or latest tomorrow. Could you try and fix the tests and make sure they pass.

**Comment by timmak** [comment]
Yep will do just ran out of time will try get done later today

File: app/addons/config/routes.js [hunk 1]
```diff
@@ -16,10 +16,10 @@ import CORSActions from "../cors/actions";
 import ClusterActions from "../cluster/cluster.actions";
 import ConfigActions from "./actions";
 import Layout from './layout';
-
+import React from 'react';
```
Comment (garrensmith) [inline app/addons/config/routes.js]: Do you need to import React here?

File: app/core/auth.js [hunk 2]
```diff
@@ -11,50 +11,30 @@
...
-  registerAuth: function (authHandlerCb) {
-    this.authHandlerCb = authHandlerCb;
-  },
-
-  registerAuthDenied: function (authDeniedCb) {
-    this.authDeniedCb = authDeniedCb;
-  },
+function authenticate(session, roles) {
+  if (session.isAdminParty()) {
+    return session.trigger('authenticated');
```
Comment (garrensmith) [inline app/core/auth.js]: if you are going to return in each if statement, you don't need the else. You can also make the `else if` just its own if statement

File: app/core/auth.js [hunk 3]
```diff
@@ -11,50 +11,30 @@
...
+  } else {
+    throw new Error('Unable to authentacte');
+  }
+}
 
-  checkAccess: function (roles) {
-    var requiredRoles = roles || [],
-    that = this;
+function authenticatoinDenied() {
+  let url = window.location.hash
```
Comment (garrensmith) [inline app/core/auth.js]: nice idea here. Could you make this a `const`

File: app/core/base.js [hunk 4]
```diff
@@ -159,4 +161,15 @@ FauxtonAPI.setSession = function (newSession) {
 
 FauxtonAPI.reducers = [];
 
+FauxtonAPI.json = (url, opts = {}) => fetch(
```
Comment (garrensmith) [inline app/core/base.js]: Could you move this into its own file, maybe something like `ajax`.  This PR https://github.com/apache/couchdb-fauxton/pull/745 had a good idea. Unfortunately it doesn't look like they will complete it. Could you rather do something similar like that for all our ajax requests with fetch

File: app/core/base.js [hunk 5]
```diff
@@ -9,9 +9,11 @@
 // WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
 // License for the specific language governing permissions and limitations under
 // the License.
-
+import "babel-polyfill";
```
Comment (garrensmith) [inline app/core/base.js]: What do we need the polyfill for?

File: app/addons/config/routes.js (hunk 1 above)
Comment (timmak) [inline app/addons/config/routes.js]:
```
configForNode: function (node) {
    ConfigActions.fetchAndEditConfig(node);
    return <Layout
      node={node}
      docURL={this.configs.documentation}
      endpoint={this.configs.url()}
      crumbs={[{ name: 'Config' }]}
      showCors={false}
    />;
  },

  configCorsForNode: function (node) {
    CORSActions.fetchAndEditCors(node);
    return <Layout
      node={node}
      docURL={this.configs.documentation}
      endpoint={this.configs.url()}
      crumbs={[{ name: 'Config' }]}
      showCors={true}
    />;
  }
```
So react needs to be imported here unless I am missing something

File: app/core/base.js (hunk 4 above)
Comment (timmak) [inline app/core/base.js]: Sure why not

File: app/core/base.js (hunk 5 above)
Comment (timmak) [inline app/core/base.js]: I need this to get test running I think promise implementation was missing I will try see I can remove requiring the whole polyfill

**Comment by timmak** [comment]
Still working on it fixing the test is turning out to be a lot harder

**Comment by garrensmith** [comment]
No problem. Let me know if I can help. You can ping me on irc `garrensmith`. Also for fast testing you can run `npm run devtests` then edit `tests/dev.js` to point to the test files you want and visit `localhost:8001` for the tests to run in the browser.

**Comment by timmak** [comment]
@garrensmith could you give this a code review ended up making a lot of changes

**Comment by garrensmith** [comment]
Sure will do. I’m traveling and then at couchdb summit. So It will take me a few days to get to it.

File: app/addons/auth/actions.js [hunk 6]
```diff
@@ -10,148 +10,162 @@
...
+  session
+    .create({
+      name: username,
+      password: password
+    })
+    .then(
+      () => {
+        FauxtonAPI.dispatch({
+          type: AUTH_CREDS_VALID,
+          options: { username: username, password: password }
```
Comment (garrensmith) [inline app/addons/auth/actions.js]: With es2015 you can make this options object look like this

```
{
 usernane,
password
}
```

Could you check if other objects could use that format as well.

File: app/addons/auth/actiontypes.js [hunk 7]
```diff
@@ -2,24 +2,24 @@
 // use this file except in compliance with the License. You may obtain a copy of
 // the License at
 //
-//   http://www.apache.org/licenses/LICENSE-2.0
+//   http =//www.apache.org/licenses/LICENSE-2.;
```
Comment (garrensmith) [inline app/addons/auth/actiontypes.js]: oops. Can you change back

File: app/addons/auth/base.js [hunk 8]
```diff
@@ -44,7 +54,7 @@ Auth.initialize = function () {
     } else if (session.isLoggedIn()) {
       link = {
         id: 'auth',
-        title: session.user().name,
+        title: session.user.name,
```
Comment (garrensmith) [inline app/addons/auth/base.js]: Is this correct? Shouldn't it be session.user().name?

File: app/addons/auth/base.js [hunk 9]
```diff
@@ -61,50 +71,16 @@ Auth.initialize = function () {
...
-    FauxtonAPI.navigate('/login?urlback=' + url, { replace: true });
-  };
-
-  FauxtonAPI.auth.registerAuth(auth);
-  FauxtonAPI.auth.registerAuthDenied(authDenied);
-};
-
-export default Auth;
+    Auth.session.fetchUser().then(function () {
+      // debugger;
```
Comment (garrensmith) [inline app/addons/auth/base.js]: Can you remove this debugging code

File: app/addons/auth/base.js [hunk 10]
```diff
@@ -61,50 +71,16 @@ Auth.initialize = function () {
...
+      // Auth.session.trigger('change');
```
Comment (garrensmith) [inline app/addons/auth/base.js]: I think we need this trigger. Could you enable it back.

File: app/addons/auth/components/LoginForm.jsx [hunk 11]
```diff
@@ -0,0 +1,101 @@
...
+class LoginForm extends React.Component {
+  constructor() {
+    super();
+    this.state = {
+      username: "",
+      password: ""
+    };
+  }
+  onInputChange(e) {
+    var change = e.target.name === "name"
```
Comment (garrensmith) [inline app/addons/auth/components/LoginForm.jsx]: Can you use `let`

File: app/addons/auth/components/LoginForm.jsx [hunk 12]
```diff
@@ -0,0 +1,101 @@
...
+          </div>
+        </div>
+      </div>
+    );
+  }
+}
+
+LoginForm.defaultProps = {
+   urlBack: "",
+  // for testing purposes only
```
Comment (garrensmith) [inline app/addons/auth/components/LoginForm.jsx]: Can you remove if its only for testing.

File: app/addons/auth/stores/ChangePassword.js [hunk 13]
```diff
@@ -0,0 +1,57 @@
+import FauxtonAPI from "../../../core/api";
+import {
+  AUTH_CLEAR_CHANGE_PWD_FIELDS,
+  AUTH_UPDATE_CHANGE_PWD_FIELD,
+  AUTH_UPDATE_CHANGE_PWD_CONFIRM_FIELD
+} from "../actiontypes";
+
+const ChangePasswordStore = FauxtonAPI.Store.extend({
```
Comment (garrensmith) [inline app/addons/auth/stores/ChangePassword.js]: If you feeling brave, you are welcome to change this all to use redux. See the permission section as an example

File: app/addons/verifyinstall/resources.js [hunk 14]
```diff
@@ -11,152 +11,159 @@
...
-        source: 'verifytestdb',
-        target: 'verifytestdb_replicate'
+        source: "verifytestdb",
+        target: "verifytestdb_replicate"
       })
     });
   },
-
-  testReplicate: function () {
+  testReplicate: function() {
```
Comment (garrensmith) [inline app/addons/verifyinstall/resources.js]: These changes don't really seem related to your PR. It just makes it a little harder to review everything

File: app/addons/auth/base.js (hunk 8 above)
Comment (timmak) [inline app/addons/auth/base.js]: user is know users setter and getters so no need for the function call

File: app/addons/auth/base.js (hunk 10 above)
Comment (timmak) [inline app/addons/auth/base.js]: There is no longer a trigger by setting the user the triggers a change

File: app/addons/auth/stores/ChangePassword.js (hunk 13 above)
Comment (timmak) [inline app/addons/auth/stores/ChangePassword.js]: will investigate this

**Comment by timmak** [comment]
Thanks @garrensmith will remove `babel-eslint` and the transform and work through the other stuff mentioned as for design changes will have a better idea once I know whether or not to do in this branch.

**Comment by justin-mcdavid-ibm** [comment]
Hi Tim, thanks for your work on this section.

Here are some ideas of what we might do with that section. 

password - inactive
password - active
admin creation - inactive
admin creation - active

We've been trying to move towards a main-nav/tabs model to free up more real estate (you can see this in the Replication section, where we have a tabs component that you could reuse).  I'm sure Garren would be able to help if you have any questions on the component.

**Comment by timmak** [comment]
@justin-mcdavid-ibm @garrensmith I will apply the designs and the redux stuff in the branch all seems straight forward

**Comment by garrensmith** [comment]
Awesome. That will be amazing. 

All misspelling thanks to my iPhone.




On Mon, Feb 13, 2017 at 4:20 AM -0500, "Tim Pinington" <notifications@github.com> wrote:












@justin-mcdavid-ibm @garrensmith I will apply the designs and the redux stuff in the branch all seems straight forward

**Comment by millayr** [comment]
hey @timmak awesome job on this.  I'd recommend taking a look at this https://github.com/apache/couchdb-fauxton/pull/840 if you haven't already for a handy tab layout component.  thanks!

**Comment by garrensmith** [comment]
I've take the work and updated it. New PR is here https://github.com/apache/couchdb-fauxton/pull/923

**Pull request closed without merging**